src/UI/Components/Diagram/Context/EventWrapper.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { useContext, useEffect } from "react";
+import { updateServiceOrderItems } from "../helpers";
+import { ActionEnum, EmbeddedEventEnum } from "../interfaces";
+import { ServiceEntityBlock } from "../shapes";
+import { CanvasContext } from "./Context";
+
+/**
+ * EventWrapper component
+ *
+ * This component is a higher-order component that wraps its children and provides event handling for all necessary communication from within JointJS code to the Composer.
+ * It uses the CanvasContext to get and set various state variables.
+ *
+ * @props {React.PropsWithChildren} props - The properties that define the behavior and display of the component.
+ * @prop {React.ReactNode} children - The children components to be wrapped.
+ *
+ * @returns {React.FC<React.PropsWithChildren>} The EventWrapper component.
+ */
+export const EventWrapper: React.FC<React.PropsWithChildren> = ({
+  children,
+}) => {
+  const {
+    setStencilState,
+    setServiceOrderItems,
+    setCellToEdit,
+    setDictToDisplay,
+    looseEmbedded,
+    setLooseEmbedded,
+  } = useContext(CanvasContext);
+
+  /**
+   * Handles the event triggered when there are loose embedded entities on the canvas.
+   * The loose embedded entities are the entities that are not connected to the main entity. If there are any, they will be highlighted and deploy button will be disabled.
```

Comment:
What with the Inter-service relations? Is the deploy disabled too when a loose Inter-service relation is dragged on the canvas?

Reply:
no, I'll create ticket for that

Reply:
Ticket link?

Reply:
https://github.com/inmanta/web-console/issues/5998